### DISCUSSION TITLE
GIthub CI. How to configure GitHubActions to be triggered only when its master branch with tag

### DISCUSSION THREAD

**Discussion by ScarletKuro** [nuke-build/nuke#950] [Q&A] [answered]
Hello,

I tied this
```CSharp
[GitHubActions(
    "release",
    GitHubActionsImage.UbuntuLatest,
    FetchDepth = 0,
    OnPushBranches = new []{ "main", "master" },
    OnPushTags = new[] { @"\d+\.\d+\.\d+" },
    PublishArtifacts = true,
    InvokedTargets = new[] { nameof(Push) },
    ImportSecrets = new[] { nameof(NuGetApiKey) })]
```
But it gets trigger also when pushed to master / main without a tag.
If we remove `OnPushBranches` then yea it will be triggered only on tag, but it can be any branch. 
As I understand Github supports something like "only" but I haven't found how to express it via GitHubActions with generating the workflow.

**Comment by ScarletKuro** [comment, marked as answer]
Okay, nvm, as I understood tags have no relation to branches, so it's not possible.

**Reply by matkoch** [reply]
Also, defining both means an OR conjunction, just as if you would set multiple branches to trigger on.

**Reply by matkoch** [reply]
You can still make a requirement in your build and check for `Repository.IsOnMainOrMasterBranch()`

**Reply by ScarletKuro** [reply]
I tried the `IsOnMainOrMasterBranch()` btw, but it's always `false` when you push a tag, because it refers to `refs/tags/` as branch.

**Reply by lahma** [reply]
@ScarletKuro maybe you could check [this publish logic](https://github.com/RicoSuter/NJsonSchema/blob/master/build/Build.Publish.cs) for some ideas. The build setup [is created here](https://github.com/RicoSuter/NJsonSchema/blob/master/build/Build.cs). I've opted to logic where in pre-stage (OnBuildInitialized) some facts are gathered and printed to console for easier tracking of  things.
